### DISCUSSION TITLE
Minimal example attempt with synthetic data

### DISCUSSION THREAD

**Discussion by dleprovost** [OHDSI/DataQualityDashboard#269] [Q&A]
Hello everyone,

I am currently trying to produce a simple example of OMOP CDM database, populated with synthetic data, and with the DataQualityDashboard executed on it. The underlying idea is to produce a tutorial draft to provide a from end-to-end minimal example of OMOP data assessment.

Here are the steps I've done:

```bash
#!/bin/bash
# Synthetic data generation using https://github.com/synthetichealth/synthea
java -jar synthea-with-dependencies.jar --exporter.csv.export true --exporter.fhir.export false -p 1000
```
```r
#!/usr/bin/R
# Create the CDM tables and load them with https://github.com/OHDSI/ETL-Synthea
devtools::install_github("OHDSI/ETL-Synthea");
ETLSyntheaBuilder::CreateSyntheaTables(connectionDetails=dbc, syntheaSchema="synthea", syntheaVersion="2.7.0");
ETLSyntheaBuilder::LoadSyntheaTables(connectionDetails=dbc, syntheaSchema="synthea", syntheaFileLoc="~/output/csv");
# The LoadSyntheaTables operation succeeded, despite repeated failures due to:
# - column "system" of relation "allergies" does not exist
# - column "start" of relation "procedures" does not exist
# - column "category" of relation "observations" does not exist
ETLSyntheaBuilder::LoadVocabFromCsv(connectionDetails=dbc, cdmSchema="omop", vocabFileLoc="output/vocabulary");
ETLSyntheaBuilder::LoadEventTables(connectionDetails=dbc, cdmSchema="omop", syntheaSchema="synthea", cdmVersion="5.3", syntheaVersion = "2.7.0");
```

So I created a new schema, `"quality"`, authorized the postgres user, then ran:
```R
devtools::install_github("OHDSI/DataQualityDashboard");
DataQualityDashboard::executeDqChecks(connectionDetails=dbc, cdmDatabaseSchema="omop", resultsDatabaseSchema="quality", cdmSourceName="SyncData");
```
And I got this output only:
```
Connecting using PostgreSQL driver
Error in rJava::.jcall(connection@jConnection, "V", "close") : 
  ignoring SIGPIPE signal
Calls: <Anonymous> -> <Anonymous> -> disconnect.default -> <Anonymous>
Execution halted
```
Nothing seems to be produced, in the filesystem nor in the database.
I did other attempts: when I set a non-existent schema as `cdmDatabaseSchema`, I rightfully got a `relation "quality.cdm_source" does not exist` error. When I put a non-existent schema as `resultsDatabaseSchema`, nothing changes and I get the same output than above.

I can't determine exactly what's failing and why. What I can change in my process, the calls, the data or the code.

Thank you all for any hints or advice you may have. 🙂

Best regards,
Damien.

**Comment by dleprovost** [comment]
I just figured the existence of the `verboseMode` parameter. 
It shows me a `connectionDetails not found` for every single check.

If I copy my connection variable in my R session to another var with this exact name, the execution succeed.

After search in the source code, and build from a modified source file, I found the bug as located in `execution.R`, line 743: `connectionDetails` is used without being passed, so it refer to a global variable. I fixed it by adding it among function parameters.

**Comment by katy-sadowski** [comment]
Resolved in https://github.com/OHDSI/DataQualityDashboard/commit/0fc4c3023058e161502b42446fb55bfa179412df
